internal/cmd/beta/public-ip/update/update.go

```diff
@@ -0,0 +1,172 @@
+			),
+			examples.NewExample(
+				`Update public IP with ID "xxx" with new labels`,
+				`$ stackit beta public-ip update xxx --labels key=value,foo=bar`,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			publicIpLabel, err := iaasUtils.GetPublicIP(ctx, apiClient, model.ProjectId, model.PublicIpId)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get public IP: %v", err)
+				publicIpLabel = model.PublicIpId
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to update public IP %q?", publicIpLabel)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("update public IP: %w", err)
+			}
+
+			return outputResult(p, model, publicIpLabel, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), associatedResourceIdFlag, "Associates the public IP with a network interface or virtual IP (ID)")
+	cmd.Flags().StringToString(labelFlag, nil, "Labels are key-value string pairs which can be attached to a public IP. E.g. '--labels key1=value1,key2=value2,...'")
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	publicIpId := inputArgs[0]
+
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	model := inputModel{
+		GlobalFlagModel:      globalFlags,
+		PublicIpId:           publicIpId,
+		AssociatedResourceId: flags.FlagToStringPointer(p, cmd, associatedResourceIdFlag),
+		Labels:               flags.FlagToStringToStringPointer(p, cmd, labelFlag),
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiUpdatePublicIPRequest {
+	req := apiClient.UpdatePublicIP(ctx, model.ProjectId, model.PublicIpId)
+
+	var labelsMap *map[string]interface{}
+	if model.Labels != nil && len(*model.Labels) > 0 {
+		// convert map[string]string to map[string]interface{}
+		labelsMap = utils.Ptr(map[string]interface{}{})
+		for k, v := range *model.Labels {
+			(*labelsMap)[k] = v
+		}
+	}
+
+	payload := iaas.UpdatePublicIPPayload{
+		NetworkInterface: iaas.NewNullableString(model.AssociatedResourceId),
```

Comment:
I think this is not the correct approach here, because even if the user does not specify `--associated-resource-id` this would send `null` and effectively disassociate the resources

Reply:
We could either remove the option to pass `--associated-resource-id` in the `update` command (and point to `associate` and `disassocaite` commands for that), or would need to handle the 3 possible cases

Reply:
I have removed the `--associated-resource-id` flag from update command. Alex also mentioned in his previous comment  :)